src/mlpack/methods/ann/ffn_impl.hpp

```diff
@@ -225,6 +225,89 @@ double FFN<OutputLayerType, InitializationRuleType, CustomLayers...>::Evaluate(
   return res;
 }
 
+template<typename OutputLayerType, typename InitializationRuleType,
+         typename... CustomLayers>
+double FFN<OutputLayerType, InitializationRuleType, CustomLayers...>::Evaluate(
+    const arma::mat& parameters, const size_t begin, const size_t batchSize)
+{
+  return Evaluate(parameters, begin, batchSize, true);
+}
+
+template<typename OutputLayerType, typename InitializationRuleType,
+         typename... CustomLayers>
+template<typename GradType>
+double FFN<OutputLayerType, InitializationRuleType, CustomLayers...>::
+EvaluateWithGradient(const arma::mat& parameters, GradType& gradient)
+{
+  double res = 0;
+  for (size_t i = 0; i < predictors.n_cols; ++i)
+    res += EvaluateWithGradient(parameters, i, gradient, 1, false);
+
+  return res;
+}
+
+template<typename OutputLayerType, typename InitializationRuleType,
+         typename... CustomLayers>
+template<typename GradType>
+double FFN<OutputLayerType, InitializationRuleType, CustomLayers...>::
+EvaluateWithGradient(const arma::mat& /* parameters */,
```

Comment:
Do you think the existing ``Gradient`` method could use the ``EvaluateWithGradient`` as well?

Reply:
@zoq It could, but there would be practically no speedup, as the same functions would be required to get called.

Reply:
Right, but we could remove some code duplication.